### PR TITLE
Add constructor to directly create QgsMultiPolygon from list of polygons

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

**Review by nirvn — APPROVED** [review]

**Comment by nyalldawson** [comment]
@nirvn I've added an equivalent for MultiLineString now. Turns out MultiPoint already has this in place.

**Review by nirvn — APPROVED** [review]
@nyalldawson , thanks, that'll make for a more consistent set of API all around. Woupidou.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1102,7 +1102,7 @@ if (WITH_CORE AND WITH_BINDINGS)
   include(SIPMacros)
 
   set(SIP_INCLUDES ${PYQT_SIP_DIR} ${CMAKE_SOURCE_DIR}/python)
-  set(SIP_CONCAT_PARTS 26)
+  set(SIP_CONCAT_PARTS 27)
```
Comment (nyalldawson) [inline CMakeLists.txt:1105]:
```suggestion
  set(SIP_CONCAT_PARTS 28)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1102,7 +1102,7 @@ if (WITH_CORE AND WITH_BINDINGS)
   include(SIPMacros)
 
   set(SIP_INCLUDES ${PYQT_SIP_DIR} ${CMAKE_SOURCE_DIR}/python)
-  set(SIP_CONCAT_PARTS 26)
+  set(SIP_CONCAT_PARTS 28)
```
Comment (nyalldawson) [inline CMakeLists.txt:1105]:
```suggestion
  set(SIP_CONCAT_PARTS 30)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -1102,7 +1102,7 @@ if (WITH_CORE AND WITH_BINDINGS)
   include(SIPMacros)
 
   set(SIP_INCLUDES ${PYQT_SIP_DIR} ${CMAKE_SOURCE_DIR}/python)
-  set(SIP_CONCAT_PARTS 26)
+  set(SIP_CONCAT_PARTS 30)
```
Comment (nyalldawson) [inline CMakeLists.txt:1105]:
```suggestion
  set(SIP_CONCAT_PARTS 22)
```

**Pull request merged**